### PR TITLE
Change README structures, Update Demo App Instructions

### PR DESCRIPTION
**Pull request by giarc3**
As the repo grows, it will become harder to keep one large README readable. Splitting into individual READMEs makes it easier to find relevant instructions.

This also updates the Android Example App README command line instructions for running with an emulator.

### CODE REVIEW

File: android/README.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+### Building
+
+From the repository root, run `android/build.sh`. This will put the compiled library and generated Java files into `android/ironoxide-android/src/main`.
+
+## Testing
+
+### Prerequisites
+
+- Successfully run `build.sh` by following the steps in [Build from Source](#build-from-source-1).
+  - This will create the java files and the `.so` files required for the `x86`, `x86_64`, and `arm64-v8a` architectures. If testing on a different architecture, you can find the Rust target to compile to [here](https://forge.rust-lang.org/release/platform-support.html).
```
Comment (cjyar) [inline android/README.md:68]: This is what `build-aar.sh` does now, right? `Build.sh` only builds one architecture.

File: android/README.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+```java
+allprojects {
+    repositories {
+        maven {
+            url "https://oss.sonatype.org/content/repositories/snapshots"
+        }
+    }
+}
+```
```
Comment (cjyar) [inline android/README.md:21]: We publish to sonatype, but that gets transferred to Maven Central. Shouldn't we direct people to the main releases? I'd expect those to be on the default repositories list already, so we might not need this note at all.

File: android/examples/Example_Application/app/build.gradle [hunk 3]
```diff
@@ -14,7 +14,7 @@ android {
 }
 
 dependencies {
-    implementation 'com.ironcorelabs:ironoxide-android:0.14.1@aar'
+    implementation 'com.ironcorelabs:ironoxide-android:0.13.1@aar'
```
Comment (cjyar) [inline android/examples/Example_Application/app/build.gradle:17]: We still need to retroactively fix `0.13.1`, right?

File: android/gradle/wrapper/gradle-wrapper.properties [hunk 4]
```diff
@@ -1,6 +1,6 @@
-#Fri Mar 20 15:26:09 MDT 2020
+#Wed Jun 24 10:49:48 MDT 2020
 distributionBase=GRADLE_USER_HOME
-distributionUrl=https\://services.gradle.org/distributions/gradle-5.4.1-all.zip
 distributionPath=wrapper/dists
-zipStorePath=wrapper/dists
 zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
```
Comment (cjyar) [inline android/gradle/wrapper/gradle-wrapper.properties:6]: This is more than a doc change. Is it going to break our use of the old `apply plugin:` syntax?

File: cpp/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# IronOxide-Cpp
+
+The building of the C++ library is currently not done by CI, so if you want to produce C/C++ bindings and the associated
```
Comment (cjyar) [inline cpp/README.md:3]: Except for the iOS library; we're doing that in CI. Is it worth mentioning that here? It'd probably just confuse things, right?

Comment (cjyar) [reply cpp/README.md:3]: Actually, we do perform the C++ build in CI to make sure it's all working. It's not build by CD and published in binary form. If we say we're not even doing CI builds, that's sort of a red flag that we don't care.

File: java/README.md [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+java.lang.System.loadLibrary("ironoxide_java")
+```
+
+### Java SDK
+
+The SDK is [published to Maven](https://search.maven.org/artifact/com.ironcorelabs/ironoxide-java).
+
+## Usage
+
+All the SDK classes can be imported from the `com.ironcorelabs.sdk` namespace.
```
Comment (cjyar) [inline java/README.md:31]: s/namespace/package/

File: android/README.md (hunk 2 above)
Comment (giarc3) [reply android/README.md]: Looks like you're right! Something I had read made me think this was required, but taking it out of the Example App has no effect, so great!

File: android/README.md (hunk 1 above)
Comment (giarc3) [reply android/README.md]: True. I think most people will only care about the x86_64 build, so I'll keep it as `build.sh` and change the wording

File: android/examples/Example_Application/app/build.gradle (hunk 3 above)
Comment (giarc3) [reply android/examples/Example_Application/app/build.gradle:17]: 0.13.1 is working.
0.13.2 and 0.14.1 are not.

File: android/gradle/wrapper/gradle-wrapper.properties (hunk 4 above)
Comment (giarc3) [reply android/gradle/wrapper/gradle-wrapper.properties:6]: True, I just wanted to make sure I was writing directions for the latest versions of everything.
No, everything was compatible, the old syntax is no longer preferred, but is still acceptable.

File: cpp/README.md (hunk 5 above)
Comment (giarc3) [reply cpp/README.md]: This file is just a rename of Colt's original `BUILDING.md`, so it is most likely out of date. 
I'll just remove this section about CI.

File: java/README.md (hunk 6 above)
Comment (giarc3) [reply java/README.md]: Done

File: android/examples/Example_Application/README.md [hunk 7]
```diff
@@ -8,16 +8,21 @@ This application provides a simple UI for encrypting/decrypting data with IronCo
...
+- An Android emulator running, or a compatible Android phone connected.
+  - To start an emulator using the command line tools, follow these steps from the folder `AndroidCLI/cmdline-tools/tools/bin`:
+    1. `./sdkmanager "emulator" "system-images;android-29;google_apis;x86_64"`
+    1. `./avdmanager create avd -n pixel_3 -k "system-images;android-29;google_apis;x86_64" -d pixel_3`
+    1. `../../../emulator/emulator -avd pixel_3 -no-snapshot -noaudio -no-boot-anim`
 
 ### Directions
 
-1. Start the AVD.
 1. From the Example_Application root, run `./gradlew installDebug`.
```
Comment (clintfred) [inline android/examples/Example_Application/README.md:26]: I didn't have ANDROID_SDK_ROOT defined which the build requires. Did I miss that step somewhere?
With `ANDROID_SDK_ROOT=/home/clintfred/Android ./gradlew installDebug` it did install properly.

File: android/README.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+To load the `ironoxide-android` library in Java, use the following:
+
+```java
+java.lang.System.loadLibrary("ironoxide_android");
+```
+
+### Authenticating
+
+To make calls, you must create a project and segment in the IronCore Admin Console. Once created, you can use a tool such as [ironoxide-cli](https://github.com/IronCoreLabs/ironoxide-cli) to create users and devices. You can then read in a device with the `DeviceContext.fromJsonString()` function and use it to initialize an IronOxide instance. An example of this is available in the [Example Application](/android/examples/Example_Application/) in "MainActivity.java".
```
Comment (clintfred) [inline android/README.md:29]: I like this pointer to `ironoxide-cli` to get started quickly, but it reads like we recommend that as a long term solution, which we don't. I don't think we need to expand a ton here, but maybe we should qualify our `ironoxide-cli` pointer a little.

File: android/examples/Example_Application/README.md (hunk 7 above)
Comment (giarc3) [reply android/examples/Example_Application/README.md:26]: Good catch, I was thinking that setting the `sdk.dir` in your local `gradle.properties` (which is in the directions) would make it all work, but I was missing Chris's fix to make that happen. Did that now, so the current directions will work correctly.

File: android/README.md (hunk 8 above)
Comment (giarc3) [reply android/README.md:29]: I see what you mean. Until we have the official JWT docs posted, I'll include a link to ironoxide's JWTClaims struct to help show what form it should take.

**Comment by coltfred** [comment]
I think this is much easier to follow. Thanks for breaking it up!

**Pull request merged**